pdns/dnsdistdist/dnsdist-tcp.cc

```diff
@@ -398,7 +398,7 @@ void IncomingTCPConnectionState::queueResponse(std::shared_ptr<IncomingTCPConnec
 
     // for the same reason we need to update the state right away, nobody will do that for us
     if (state->active()) {
-      updateIO(state, iostate, now);
+      state->updateIO(state, iostate, now);
```

Comment:
Is it really necessary to call a method on `state` while also passing `state` as arg?  Which object plays which role? Is theer a way to simplify this and make it less ambiguous? Also, reading the code is a bit hard as the name `state` is used as a name for a `std::shared_ptr<IncomingTCPConnectionState>` and also for a `IOState`. I know this is already in the existing code.

Reply:
done